nimble/host/src/ble_hs_mbuf.c

```diff
@@ -19,6 +19,20 @@
 
 #include "host/ble_hs.h"
 #include "ble_hs_priv.h"
+#include "nimble/hci_common.h"
+
+#if MYNEWT_VAL(BLE_HS_OWN_MBUF_FOR_HEADERS)
+struct os_mbuf_pool ble_hs_hdr_mbuf_pool;
+struct os_mempool ble_hs_hdr_pool;
+
+#define BLE_HS_MEMPOOL_BLOCK_SIZE  (BLE_ATT_PREP_WRITE_CMD_BASE_SZ +\
+                                    BLE_HCI_DATA_HDR_SZ + BLE_L2CAP_HDR_SZ)
+static os_membuf_t ble_hs_hdr_mem[
+    OS_MEMPOOL_SIZE(MYNEWT_VAL(BLE_HS_OWN_MBUF_FOR_HEADERS_CNT),
+                    BLE_HS_MEMPOOL_BLOCK_SIZE)
+];
+
```

Comment:
Not that I have given this a ton of thought but I wonder if another way to do this is a have a very small mbuf pool added to the msys pools for nimble? This would make for less code but could still waste bib mbufs if no small ones are available. I realize this last comment is probably lame but too bad we cannot force the applications to reserve some space so we can add these headers to the existing mbufs as it would save other overhead.

Reply:
I was thinking to have msys_init_2 with smaller block size but then I found out that there are many places in the system where `os_msys_get_pkthdr(0,0)` is used which means, our smaller mempool whould be used very quickly.

That is why for the moment let's make sure this is only for NimBLE in this way. Does it makes sense?